### ISSUE TITLE
export doesn't work inside anonymous classes

### ISSUE THREAD

**Issue by soronpo** [lampepfl/dotty#7920]
## minimized code

```Scala
class Bar {val glass : Int = 0}
class Foo(val bar : Bar) 
val f = new Foo(new Bar) {
  export bar._
}
val fGlass = f.glass //error: value glass is not a member of Foo
```
## expectation
This code should compile. I see no reason why export shouldn't work in anonymous classes.

**Comment by odersky** [comment]
This has nothing to do with exports. If you replace the export with `export bar._`, it fails as well. See 
https://contributors.scala-lang.org/t/better-type-inference-for-scala-send-us-your-problematic-cases/2410/73 for background.

I believe we should implement the `Structural` idea that @smarter proposed in that thread, though.

**Comment by soronpo** [comment]
Fixed in #9201 
```scala
class Bar {val glass : Int = 0}
class Foo(val bar : Bar) extends scala.reflect.Selectable
val f = new Foo(new Bar) {
  export bar._
}
val fGlass = f.glass
```
https://scastie.scala-lang.org/IrMU1eYaSBOW33nPbo2JOQ
